switch_model/utilities/__init__.py

```diff
@@ -7,24 +7,54 @@
 from __future__ import print_function
 
 import os, types, importlib, re, sys, argparse, time, datetime
-import __main__ as main
+import switch_model.__main__ as main
 from pyomo.environ import *
+from switch_model.utilities.scaling import _ScaledVariable, _get_unscaled_expression
 import pyomo.opt
 
 # Define string_types (same as six.string_types). This is useful for
 # distinguishing between strings and other iterables.
-try:
-    # Python 2
-    string_types = (basestring,)
-except NameError:
-    # Python 3
-    string_types = (str,)
+string_types = (str,)
 
 # Check whether this is an interactive session (determined by whether
 # __main__ has a __file__ attribute). Scripts can check this value to
 # determine what level of output to display.
 interactive_session = not hasattr(main, '__file__')
 
+
+class _AbstractModel(AbstractModel):
```

Comment:
This looks good! Do you think of this as a general solution that we could add on top of it or just a quick solution. I am thinking if we require to modify the abstract model in the future it might be worth our time to find a general approach.

Reply:
I see this as a general solution! We can simply add to `_AbstractModel` if we want to make custom changes to Pyomo's abstract model.